src/hpr_routing.erl

```diff
@@ -82,16 +82,25 @@ deliver_packet(Packet, StreamHandler, [Route | Routes]) ->
         [hpr_route:lns(Route)]
     ),
     Protocol = hpr_route:protocol(Route),
-    %% FIXME: delivery could be halted to multiple routes if one of the earlier
-    %% Protocol:send(...) errors out.
+
     Resp =
-        case Protocol of
-            router ->
-                hpr_protocol_router:send(Packet, StreamHandler, Route);
-            gwmp ->
-                hpr_gwmp_router:send(Packet, StreamHandler, Route);
-            _OtherProtocol ->
-                lager:warning([{protocol, _OtherProtocol}], "unimplemented"),
+        try
+            case Protocol of
+                router ->
+                    hpr_protocol_router:send(Packet, StreamHandler, Route);
+                gwmp ->
+                    hpr_gwmp_router:send(Packet, StreamHandler, Route);
+                _OtherProtocol ->
+                    lager:warning([{protocol, _OtherProtocol}], "unimplemented"),
+                    ok
+            end
+        catch
+            Class:Exception:Stacktrace ->
```

Comment:
Is `Stacktrace` the correct exception to catch?  This seems brittle if ever stacktraces get disabled (e.g., BEAM setting for performance).

Mind you, I'm not well versed in Erlang's exceptions... so not a show-stopper.

Reply:
Interesting question. Worth looking into.
If disabling stacktraces is a runtime thing, then the compiler should do something reasonable with code like this.

Reply:
The `system_flag` `backtrace_depth` limits the depth of stack captured in exceptions and returned by `process_info` on `current_stacktrace`. Is this what you're referring to? If this is set to `0`, you'd still get a `Stacktrace` value, but it will be an empty list.
Also, given the API for raising an error (where the stack trace is mandatory), I suspect there is always a `Stacktrace` value when catching an exception.
Doing anything else would make source code dependent on runtime settings, which is generally avoided.

Reply:
Thanks for the clarification, @marcsugiyama .
I expected something like that.